infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/BrokerEnvironmentFactory.java

```diff
@@ -247,7 +247,25 @@ private String generateUniqueName(String suffix) {
    */
   @VisibleForTesting
   protected String generateContainerNameFromImageRef(String image) {
-    return image.toLowerCase().replaceAll("[^/]*/", "").replaceAll("[^\\d\\w-]", "-");
+    String containerName;
```

Comment:
While this is certainly improving the situation, it is still not taking care of the possible clashes. What do you think can be done there?

Reply:
A container name can be up to 30 chars, if you want to use an image with a *really* long tag there's not much we can do that's intelligent. Apart from that we could choose a completely different tack for naming these containers, but that's potentially more risky and I would prefer that be a separate issue + PR to minimize risk in 7.9.x.

Are there any big cases I've missed in the PR? I admit I wrote it hastily.

Reply:
One solution would be to just name the containers "metadata-broker" and "artifacts-broker", though that obscures the version being used in the default use case (where it's currently something like `che-plugin-metadata-broker-v3-1-0`.